services/secrets/src/api.js

```diff
@@ -50,10 +57,25 @@ builder.declare({
 }, async function(req, res) {
   const { name } = req.params;
   const { secret, expires } = req.body;
-  await this.db.fns.upsert_secret(name, this.db.encrypt({
-    value: Buffer.from(JSON.stringify(secret), 'utf8'),
-  }), new Date(expires));
-  res.reply({});
+
+  try {
+    const [item] = await this.db.fns.get_secret(name);
+
+    await this.db.fns.upsert_secret(name, this.db.encrypt({
+      value: Buffer.from(JSON.stringify(secret), 'utf8'),
+    }), new Date(expires));
+
+    await this.db.fns.insert_secrets_audit_history(
+      name,
+      'secret',
+      await req.clientId(),
+      item === undefined ? AUDIT_ENTRY_TYPE.SECRET.CREATED : AUDIT_ENTRY_TYPE.SECRET.UPDATED
+    );
+
+    res.reply({});
+  } catch (e) {
+    throw e;
```

Comment:
this is probably redundant if we don't send custom response
Maybe we don't need try/catch here at all, as api has an error-catch middleware